### ISSUE TITLE
508-defect-3: Error message needs to tell the user why the telephone number is invalid

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#57285]
### Point of contact

@jeana-adhoc

### Severity level

3, Moderate. Should be fixed in 1-3 sprints post-launch.

### Details

The 508 Office contacted me and said the telephone error message needs to be more clear and the user needs to be told why the phone number is invalid.

### Reproduction steps

Reproduction steps
1. Navigate to the "[Your Contact information](https://www.va.gov/housing-assistance/disability-housing-grants/apply-for-grant-form-26-4555/contact-information-2)" page on the 26-4555 form
2. Tab through the phone number field (Don't enter anything in the field.)
3. The error message reads "Please enter a phone number."
4. Now, enter 2 numbers in the phone number field.
5. The error message reads "Please enter a 10-digit phone number (with or without dashes)" The 508 office would like this second message to also be the message if the field is left blank.

### Proposed solution or next steps

While the current implementation follows the guidance in the [design system](https://design.va.gov/patterns/ask-users-for/phone-numbers), we need to have more specific language in the error message.

When there is no phone number entered, the error message should read:
> Please enter a 10-digit phone number (with or without dashes)

This change should _only_ impact our forms, and should not edit the default message in the component itself (forms library) at this time.

### References, articles, or WCAG support

[WCAG 3.3.3. Error suggestions](https://www.w3.org/TR/UNDERSTANDING-WCAG20/minimize-error-suggestions.html#:~:text=Specific%20Benefits%20of%20Success%20Criterion,and%20how%20to%20correct%20it.)

### Type of issue

- [X] Screenreader
- [ ] Keyboard
- [ ] Focus
- [ ] Headings
- [x] Color
- [ ] Zoom
- [ ] Semantics
- [ ] Axe-core
- [X] Something else
